src/material/slider/slider-input.ts

```diff
@@ -102,6 +102,7 @@ export class MatSliderThumb implements _MatSliderThumb, OnDestroy, ControlValueA
     this._updateThumbUIByValue();
     this._slider._onValueChange(this);
     this._cdr.detectChanges();
```

Comment:
oh, so it calls `detectChanges` here! Does it effectively mean CD runs twice, if the value is set during change detection?

looks like other material input components (`matInput` directive, `mat-radio`) do not do this.

I suspect it might be a source of subtle issues down the line

Reply:
From what I remember, this call to `detectChanges` is needed in order to avoid changed after checked errors. This function is only called when the slider value is set manually so the additional round of change detection shouldn't be too expensive. From what I understand about other components, they don't do this because they do not use the underlying input as the source of truth and therefore don't need to handle changed after checked errors